modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/result/display/builder/SearchResultSummaryDisplayBuilder.java

```diff
@@ -682,6 +693,17 @@ protected void buildModelResource(
 		String modelResource = _resourceActions.getModelResource(
 			_themeDisplay.getLocale(), className);
 
+		if (className.startsWith(ObjectDefinition.class.getName() + "#")) {
+			String[] parts = StringUtil.split(className, "#");
+
+			ObjectDefinition objectDefinition =
+				_objectDefinitionLocalService.fetchObjectDefinition(
+					Long.valueOf(parts[1]));
+
+			modelResource = objectDefinition.getLabel(
+				_themeDisplay.getLocale());
```

Comment:
@BryanEngler Please use the `get` method instead of the `fetch` method if you aren't going to handle the possibility of the returned object being null. That will make the error message a lot more meaningful if the objectDefintionId doesn't exist.

In my case, I'm currently unable to suggest a workaround to the customer because I don't know what the bad objectDefinitionId is. If I had this information, I could've told them to blocklist the className of the non-existent ObjectDefinition.